### ISSUE TITLE
[bug] `no-unnecessary-key` false-positive with array in parent

### ISSUE THREAD

**Issue by silverwind** [Rel1cx/eslint-react#1436]
### Describe the bug

Maybe a bit of a contrived example (because `getChild` could be converted to a regular react component) but still:

```jsx
export default function App() {
  return [getChild()];
}

function getChild() {
  return <div key="key">foo</div>; // Error: Unnecessary `key` prop on this element. The `key` prop is not needed outside of dynamic rendering contexts
}
```

`key` seems necessary here to avoid React warning `Warning: Each child in a list should have a unique "key" prop"` as seen in the reproduction link.

### Reproduction

https://playcode.io/react-playground--019c0528-843d-769b-93f7-387ad3bae263
